### PR TITLE
⭐️ github app authentication

### PR DESCRIPTION
**Pull request by imilchev**
Allow scanning github with app authentication:
```
cnquery scan github org mondoohq --app-id 1234 --app-installation-id 123456 --app-private-key <path-to-pem>
```

### CODE REVIEW

File: providers/github/provider/provider.go [hunk 1]
```diff
@@ -46,17 +46,35 @@ func (s *Service) ParseCLI(req *plugin.ParseCLIReq) (*plugin.ParseCLIRes, error)
...
+	isAppAuth := false
+	if appId, ok := req.Flags[connection.OPTION_APP_ID]; ok && len(appId.Value) > 0 {
+		conf.Options[connection.OPTION_APP_ID] = string(appId.Value)
+
+		installId := req.Flags[connection.OPTION_APP_INSTALLATION_ID]
+		conf.Options[connection.OPTION_APP_INSTALLATION_ID] = string(installId.Value)
+
+		pk := req.Flags[connection.OPTION_APP_PRIVATE_KEY]
+		conf.Options[connection.OPTION_APP_PRIVATE_KEY] = string(pk.Value)
+		isAppAuth = true
```
Comment (chris-rock) [inline providers/github/provider/provider.go:62]: If the app-id is set but installation id or private key are missing we should return an error

Comment (imilchev) [reply providers/github/provider/provider.go:62]: that's done in the connection, not here. I wanted to not duplicate the logic for checking if everything is set in the `ParseCLI`

Comment (chris-rock) [reply providers/github/provider/provider.go:62]: That makes a lot of sense.

**Review by chris-rock — APPROVED** [review]
Thank you @imilchev

**Pull request merged**